myapp/src/pages/LoginPage.js

```diff
@@ -0,0 +1,86 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import BeatLoader from "react-spinners/ClipLoader";
+
+import InputField from "../components/InputField";
+
+function LoginPage() {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+
+  const [usernameError, setUsernameError] = useState("");
+  const [passwordError, setPassswordError] = useState("");
+
+  let [loading, setLoading] = useState(false);
+
+  const navigate = useNavigate();
+
+  function setData(event) {
+    event.preventDefault();
+
+    setUsernameError("");
+    setPassswordError("");
+
+    if (username === "") {
+      setUsernameError("Username is required!");
+      return;
+    } else if (password === "") {
+      setPassswordError("Password is required!");
+      return;
+    }
+
+    setLoading(!loading);
+    const timeout = setTimeout(() => {
+      navigate("/home", { state: { username } });
+    }, 1500);
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }
```

Comment:
We should look to take advantage of arrow functions in this case, and make use of react state hooks (useMemo, useCallback etc). State hooks will help in managing the state of a component more effectively with minimum conflicts and unnecessary rendering.

Reply:
Convert all functions to arrow functions and used the useMemo, useCallback hooks.
aa439fed25ecd2380580da968e5ee9fca0a39698